### ISSUE TITLE
[Help needed or Bug] ? Fails to create test connection alias at test suite start

### ISSUE THREAD

**Issue by liqueurdetoile** [vierge-noire/cakephp-test-suite-light#7]
CakePHP version : 4.2.2
Suite version :  2.1.1

I ran into an issue when switching to this test suite and launched tests. It throws me `Cannot create alias of "test" as it does not exist.`

All was running fine in classic cakephp fixtures setup.

After some investigations, it appears that the `FixtureManager::_initDb` call, that in turn calls the faulty `FixtureManager::_aliasConnections` is done in your manager constructor. It's before test bootstrap and far earlier than cakephp does, therefore datasources configuration is not already loaded and alias request fails.

As it is also natively called by `FixtureInjector::fixturize` in test start hook, simply removing the constructor of your `FixtureManager` solves my issue.

Anyway, I still don't know if I missed something in the suite setup or if it's a bug.

<details>
<summary>Here's my phpunit config file</summary>

Test bootstrap hook is done in the phpunit header.
```
<?xml version="1.0" encoding="UTF-8"?>
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" colors="true" processIsolation="false" stopOnFailure="false" bootstrap="tests/bootstrap.php" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
  <coverage>
    <include>
      <directory suffix=".php">src/</directory>
      <directory suffix=".php">plugins/*/src/</directory>
    </include>
    <exclude>
      <file>src/Console/Installer.php</file>
    </exclude>
  </coverage>
  <php>
    <ini name="memory_limit" value="-1"/>
    <ini name="apc.enable_cli" value="1"/>
    <env name="UNIT_TESTING" value="true" force="true"/>
  </php>
  <!-- Add any additional test suites you want to run here -->
  <testsuites>
    <testsuite name="app">
      <directory>tests/TestCase/</directory>
    </testsuite>
    <!-- Add plugin test suites here. -->
  </testsuites>
  <!-- Setup a listener for fixtures -->
  <listeners>
    <!-- <listener class="Cake\TestSuite\Fixture\FixtureInjector"> -->
    <listener class="CakephpTestSuiteLight\FixtureInjector">
      <arguments>
        <!-- <object class="Cake\TestSuite\Fixture\FixtureManager"/> -->
        <object class="CakephpTestSuiteLight\FixtureManager"/>
      </arguments>
    </listener>
  </listeners>
  <!-- Ignore vendor tests in code coverage reports -->
</phpunit>
```
</details>

Anyway, thanks a lot for offering an alternate suite and fixtures handling 😄

**Comment by pabloelcolombiano** [comment]
Hi @liqueurdetoile thanks for the hint. I'll have a look today or tomorrow, there is indeed a fix needed there.

**Comment by pabloelcolombiano** [comment]
This should now be fixed. There was indeed no reason to call the _iniDB method in the constructor.

Thanks for the hint and for using the package. Have fun with the test suite light and the fixture factories!

**Comment by liqueurdetoile** [comment]
Thanks for taking a look into it. Problem solved for connection alias error 😄 

Still a little issue though : `PHP Warning:  Use of undefined constant CONFIG - assumed 'CONFIG' (this will throw an Error in a future version of PHP) in [...]/vendor/cakephp/cakephp/src/Core/Configure/Engine/PhpConfig.php on line 66'

It's also coming from `FixtureManager::loadConfiguration` using `CONFIG` constant before bootstrap file is loaded by PHPUnit. I've seen [in your test bootstrap](https://github.com/vierge-noire/cakephp-test-suite-light/blob/c839bf6ced7750a8405ceb08d4989bd1c4ac9761/tests/bootstrap.php#L49) that CONFIG is settled before creating a new FixtureManager.

Is there a way to tell PHPUnit to load bootstrap before FixtureManager ?

On the other hand, `FixtureManager::loadConfiguration` is deprecated and can maybe also be simply removed from constructor ?

**Comment by pabloelcolombiano** [comment]
Thanks for the hint again. I agree that we should simply get rid of the deprecated `loadConfiguration` method.

**Comment by pabloelcolombiano** [comment]
This should now all be solved with v2.1.3 reps. v1.1.3.

Let me know if you see an issue somewhere here or there.
